### ISSUE TITLE
Feature request: Reflection capabilities on registered Polymer elements

### ISSUE THREAD

**Issue by niflostancu** [Polymer/polymer#2152]
A thought occurred to me recently: it would be nice to be able to reflect/extend others' custom elements at runtime (i.e. by adding our own behaviors without modifying the element's code).

It has some neat use cases: as a workaround for Polymer 0.5's _extends on custom elements_ feature, for modifying third party elements' functionality etc.

So I went ahead and implemented a Polymer extension library that does just this: [niflostancu/polymer-reflection](https://github.com/niflostancu/polymer-reflection). 

But it has one problem: it uses several private methods of Polymer (which I intercepted and altered), which may break at any time in the future. 

It would be nice to have some official API for:
- querying the prototype of a registered element;
- dynamically adding/removing behaviors to/from an already-registered custom element.

Basically, some sort of official Polymer extension mechanism.
Or just an advanced artificial intelligence algorithm that automatically reads the developer's mind and makes everything possible :D 

Thanks.

**Comment by kevinpschaaf** [comment]
Our plan is to support proper custom element inheritance. https://www.polymer-project.org/1.0/docs/#inheritance

This would be a different model of element extension; we'll keep this issue here to consider when we tackle inheritance.  Thanks for the feedback.

**Comment by markhats** [comment]
Any timeline on inheritance?

**Comment by devinivy** [comment]
@niflostancu nice library!  I attacked this problem from the other end, adding a `toBehavior()` method to Polymer components which allows you to use a component as a behavior in order to extend it.  It's in a project called [misbehave](https://github.com/devinivy/misbehave).

```html
<link rel="import" href="../polymer/polymer.html">
<link rel="import" href="../misbehave/misbehave.html">
<link rel="import" href="base-component.html">

<script>
  Polymer({

    is: 'ext-component',

    // ext-component now effectively extends the Polymer component base-component
    behaviors: [
      document.createElement('base-component').toBehavior()
    ]

  });
</script>
```

**Comment by TimvdLippe** [comment]
You can now properly extend an existing custom element with your own behavior. Read up on https://www.polymer-project.org/2.0/docs/devguide/custom-elements#extending-elements for more information. (For hybrid elements, you can do `class MyElement extends customElements.get('paper-button')`
